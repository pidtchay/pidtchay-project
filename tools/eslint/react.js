module.exports = {
    'react/no-did-update-set-state': 0,
    'react-hooks/exhaustive-deps': 0,
    'react/state-in-constructor': [2, 'never'],
    'react/destructuring-assignment': 0,
    'react/no-danger': 2,
    'react/prefer-stateless-function': 1,
    'react/no-array-index-key': 1,
    'react/jsx-props-no-spreading': 0,
    'react/static-property-placement': ['warn', 'property assignment', { // Указывает, где должны быть размещены статические свойства компонента React
      'childContextTypes': 'static getter',
      'contextTypes': 'static public field',
      'contextType': 'static public field',
      'displayName': 'static public field',
    }],
    'react/require-default-props': 1,
    'react/sort-comp': [1, { order: ['static-methods', 'lifecycle', 'everything-else', 'render']}],

}