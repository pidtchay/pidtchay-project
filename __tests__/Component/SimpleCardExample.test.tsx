import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Card } from 'antd';

function SimpleCard () {
  return (
        <div>
            <Card title='Default size card' extra={<a href='#'>More</a>} style={{ width: 300 }}>
                <p>Card content</p>
                <p>Card content</p>
                <p>Card content</p>
            </Card>
        </div>
  );
}

describe('SimpleCard', () => {
  it('snapshot', () => {
    const component = renderer.create(<SimpleCard />);
    expect(component).toMatchSnapshot();
  });
});