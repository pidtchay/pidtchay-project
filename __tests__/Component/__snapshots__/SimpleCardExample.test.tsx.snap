// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleCard snapshot 1`] = `
<div>
  <div
    className="ant-card ant-card-bordered"
    style={
      Object {
        "width": 300,
      }
    }
  >
    <div
      className="ant-card-head"
      style={Object {}}
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Default size card
        </div>
        <div
          className="ant-card-extra"
        >
          <a
            href="#"
          >
            More
          </a>
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <p>
        Card content
      </p>
      <p>
        Card content
      </p>
      <p>
        Card content
      </p>
    </div>
  </div>
</div>
`;
